lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -1040,16 +1073,17 @@ def payment_details(payment_method, options = {})
         when String
```

Comment:
Can we remove this `case` statement, since we're now only dealing with `payment_method`s as object/hashes, or am I confusing things.

Reply:
I think the reason we check for a string is that thirdpartytokens can be strings.